orm/versioning_bucket.go

```diff
@@ -31,19 +54,51 @@ func (b VersioningBucket) GetLatestVersion(db weave.ReadOnlyKVStore, id []byte)
 		var vID VersionedIDRef
 		idData := m.Key[dbKeyLength:]
 		if err := vID.Unmarshal(idData); err != nil {
-			return nil, errors.Wrap(err, "wrong key type")
+			return idWithoutVersion, nil, errors.Wrap(err, "wrong key type")
 		}
 		if vID.Version > highestVersion.Version {
 			highestVersion = vID
 			found = m
 		}
 	}
-	if len(found.Value) == 0 {
-		return nil, errors.Wrap(errors.ErrNotFound, "unknown id")
+	if len(found.Key) == 0 {
+		return idWithoutVersion, nil, errors.Wrap(errors.ErrNotFound, "unknown id")
+	}
+	if tombstone.Equal(found.Value) {
+		return idWithoutVersion, nil, errors.ErrDeleted
+	}
+	obj, err := b.Parse(found.Key, found.Value)
+	if err != nil {
+		return idWithoutVersion, nil, err
+	}
+	return highestVersion, obj, err
+}
+
+// Get works with a marshalled VersionedIDRef key. Direct usage should be avoided in favour of
+// GetVersion or GetLatestVersion.
+// Unlike the classic Get function it returns:
+//  - ErrNotFound when not found
+//  - ErrDeleted when deleted
+// Object won't be nil in success case
```

Comment:
Do we/should we update all other Get to return ErrNotFound instead of nil?
(eg. create an issue for this if you look at other packages and see something needing fixing)